### DISCUSSION TITLE
Audio ducking for streams when team radio is used on onboard streams

### DISCUSSION THREAD

**Discussion by Petarkco** [f1multiviewer/issue-tracker#18] [Feature requests] [answered]
Basically if you watch the World Feed or Pit Lane Channel with the commentary on at the same time as watching onboard streams it would be nice to have the commentary faded down when team radio is head on the onboard streams.

This would allow people to watch with commentary and not have to worry about not hearing team radio from drivers they are following properly.

**Comment by JustJoostNL** [comment]
Woww! This is a great idea, I would use this for sure!

**Comment by Psybertr0n** [comment]
I think this feature may be a bit too difficult, perhaps even impossible, to implement with the available API as it requires processing live audio in a DRM protected stream and detecting if the audio goes over a certain peak so that it mutes the main feed audio. 

At present a "hack" for this would be to have your onboard streams set to provide audio through a separate device as the main feed. For example you could wear wireless earphones connected to your PC, have the main feed audio coming out of your speakers and the onboard stream audio through your wireless earphones. It's not ideal and the constant engine sound in your ears will be annoying but it's perhaps the only way to not miss driver audio.

**Comment by f1multiviewer** [comment]
I've checked, and while APIs are available to analyse audio (https://developer.mozilla.org/en-US/docs/Web/API/AnalyserNode), it would not be possible for DRM-protected streams, as audio will return `-Infinity` for all values.

**Comment by f1multiviewer** [comment, marked as answer]
Actually, we now have something like this, as it seems FOM (for now) is streaming 2-channel audio for OBCs (team radio audio track), one has *way* less engine noise. In v1.5.2, you'll be able to press `P` (and see in the debug menu with `D`) your pan direction change. It's not the same for each OBC, but you can easily figure it out. Pressing it multiple times toggles between left, right and center.

**Reply by interi-mhickman** [reply]
I just came to ask for left/right/center pan for team radio. Stoked that it's already available.

**Reply by interi-mhickman** [reply]
So it doesn't actually pan the audio, which is what I was looking for. I want to hear one driver's audio mixed 100% left channel, another mixed all right channel. It makes it easier to follow multiple conversations. When RaceControl was working, I used mpv's audio filter to do this:
Left pan: `mpv --af='pan=stereo|c0<c0+c1'`
Right pan: `mpv --af='pan=stereo|c1<c0+c1'`
Can something similar be achieved with MultiViewer?

**Reply by f1multiviewer** [reply]
@interi-mhickman I think so, could you create a separate discussion for that?
